packages/core/src/agent/__tests__/Agent.test.ts

```diff
@@ -195,7 +196,35 @@ describe('Agent', () => {
       expect(container.resolve(MessageSender)).toBe(container.resolve(MessageSender))
       expect(container.resolve(MessageReceiver)).toBe(container.resolve(MessageReceiver))
       expect(container.resolve(Dispatcher)).toBe(container.resolve(Dispatcher))
+      expect(container.resolve(FeatureRegistry)).toBe(container.resolve(FeatureRegistry))
       expect(container.resolve(EnvelopeService)).toBe(container.resolve(EnvelopeService))
     })
   })
+
+  it('all core features are properly registered', () => {
+    const agent = new Agent(config, dependencies)
+    const registry = agent.dependencyManager.resolve(FeatureRegistry)
+
+    const protocols = registry.query({ featureType: 'protocol', match: '*' }).map((p) => p.id)
+
+    expect(protocols).toEqual(
```

Comment:
Can we maybe add a length check here? That way if new protocols are added the test will fail, forcing you to update it with the new protocol uri.

Reply:
Good idea. This test will force us to always double-check the modules we register.